src/main.js

```diff
@@ -30,32 +35,44 @@ var scrapeIndividualProduct = function (page) {
   };
 };
 
-var onProductListScrapeEnd = function () {
-  require('fs').write('out/products-no-metadata.json', JSON.stringify(products), 'w');
-  	
+var onProductListScrapeEnd = function() {
+  fs.write('out/products-no-metadata.json', JSON.stringify(products), 'w');
+
   // create a 10-page pool
   var pagePool = new PagePool(10);
-  
-  // ...and fire on all cylinders! 
-  Q.all(products.map(scrapeIndividualProduct(pagePool))).then(function (ps) {
+
+  // ...and fire on all cylinders!
+  Q.all(products.map(scrapeIndividualProduct(pagePool))).then(function(ps) {
     pagePool.destroy();
     require('fs').write('out/products-metadata.json', JSON.stringify(ps), 'w');
+    fs.write('out/products-metadata.json', JSON.stringify(ps), 'w');
+
     phantom.exit();
   });
 };
 
 s.on('error', dumpAndExit);
 
-// when the product list scrape is complete, write out the products
-// without metadata and then scrape the metadata.
-s.on('end', onProductListScrapeEnd);
-
-s.on('product', function (p) {
-  products.push(p);
+s.on('product', function(product) {
+  products.push(product);
 });
 
-s.on('newPage', function (p) {
-  console.log('New Page: ' + p);
+s.on('newPage', function(pageNumber) {
+  console.log('New Page: ' + pageNumber);
 });
 
-s.scrape();
\ No newline at end of file
+if (system.args.indexOf('--with-metadata') !== -1) {
```

Comment:
I added this check so it will not do the full dump unless you send in the `--with-metadata` flag when running.